### PR TITLE
Add recovery in progress error

### PR DESCRIPTION
**Pull request by Connor1996**
When doing online recovery, we'd like the region to reject writes with the error `RecoveryInProgress`.

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: proto/errorpb.proto [hunk 1]
```diff
@@ -128,6 +128,11 @@ message DataIsNotReady {
     uint64 safe_ts = 3;
 }
 
+message RecoveryInProgress {
+    // The requested region ID
+    uint64 region_id = 1;
```
Comment (BusyJay) [inline proto/errorpb.proto:133]: Isn't it part of the request?

Comment (Connor1996) [reply proto/errorpb.proto:133]: nope, the request has the context, but the response doesn't.

Comment (BusyJay) [reply proto/errorpb.proto:133]: What I mean client should know the request corresponding to the response, so it should know the region id.

Comment (Connor1996) [reply proto/errorpb.proto:133]: Yes, but seems the error `NotLeader`, `RegionNotFound` and etc. includes the region id as well

**Review by BusyJay — APPROVED** [review]

File: proto/errorpb.proto [hunk 2]
```diff
@@ -148,4 +153,6 @@ message Error {
     DataIsNotReady data_is_not_ready = 13;
     RegionNotInitialized region_not_initialized = 14;
     DiskFull disk_full = 15;
+    // Online recovery is still in performing, reject writes to avoid potential issues
+    RecoveryInProgress RecoveryInProgress = 16;
```
Comment (overvenus) [inline proto/errorpb.proto:157]: Why not use snake_case? For the sake of consistency 😅

Comment (Connor1996) [reply proto/errorpb.proto:157]: oh, miswrite

**Pull request merged**